main/src/library/Path.flix

```diff
@@ -126,6 +126,29 @@ namespace Path {
         javaFile.canWrite()
     })
 
+    ///
+    /// Read the specified path into an array of bytes
+    ///
+    pub def readBuffer(p: Path): Result[Array[Int8], IOError] & Impure = try {
+        use InputStream.{available, Error, readBuffer};
+
+        import new java.io.FileInputStream(##java.io.File) as newFileInputStream;
+
+        let Path(javaFile) = p;
+        let javaInputStream = newFileInputStream(javaFile);
+        let inputStream = InputStream(javaInputStream as ##java.io.InputStream);
+        inputStream.available()
+            |> Result.flatMap(a -> {
+                    let arr = [0 as Int8; a];
+                    let read = readBuffer(arr, inputStream);
+                    Result.map(_ -> arr, read)
+                })
+            |> Result.mapErr(_ -> Generic("InputStream error (swallowed)"))
```

Comment:
This is one of the cases where a `Result.mapErr` method is invaluable. Moving to a unified error type for functions that need ExceptionHandling would be an option to circumvent the need in this, however.

Reply:
Okay, but it should at least propagate the error message, otherwise its pretty useless.